test/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.test.cpp

```diff
@@ -0,0 +1,358 @@
+    Size rowCount = referenceData.getRowCount();
+
+    {
+        const NRLMSISE00Public nrlmsise = {};
+
+        for (Index i = 0; i < rowCount; i++)
+        {
+            const DateTime datetime =
+                DateTime::Parse(referenceData(i, "DATE").accessString(), DateTime::Format::Standard);
+
+            Instant instant = Instant::DateTime(datetime, Scale::UTC);
+
+            const LLA lla = LLA(Angle::Degrees(0), Angle::Degrees(0), Length::Meters(500000));
+
+            NRLMSISE00Public::nrlmsise_input input;
+            NRLMSISE00Public::ap_array aph;
+
+            nrlmsise.computeNRLMSISE00Input(input, aph, lla, instant);
+
+            EXPECT_TRUE(referenceData(i, "DAILY_FLUX").accessReal().isNear(input.f107, 0.001)) << String::Format(
+                "{} != {} on {}", referenceData(i, "DAILY_FLUX").accessReal(), input.f107, datetime.toString()
+            );
+            EXPECT_TRUE(referenceData(i, "AVG_FLUX").accessReal().isNear(input.f107A, 0.001)) << String::Format(
+                "{} != {} on {}", referenceData(i, "AVG_FLUX").accessReal(), input.f107A, datetime.toString()
+            );
+        }
+    }
+}
+
+TEST(OpenSpaceToolkit_Physics_Environment_Atmospheric_Earth_NRLMSISE00, GetDensityAtOrekit3HrMarksPreciseLST)
+{
+    const File referenceDataFile = File::Path(Path::Parse(
+        "/app/test/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00/OreKitNRLMSISE500km3HourMarks.csv"
+    ));
+
+    const Table referenceData = Table::Load(referenceDataFile, Table::Format::CSV, true);
+    Size rowCount = referenceData.getRowCount();
+
+    const SPICE spiceSun = {SPICE::Object::Sun};
+
+    const Shared<const Frame> sunFrameSPtr = spiceSun.accessFrame();
+
+    {
+        const NRLMSISE00 nrlmsise = {};
+
+        for (Index i = 0; i < rowCount; i++)
+        {
+            const Real latitude = referenceData(i, "LAT").accessReal();
+            const Real longitude = referenceData(i, "LON").accessReal();
+            const Real altitude = referenceData(i, "ALT").accessReal();
+            const Real referenceDensity = referenceData(i, "DENSITY").accessReal();
+            const DateTime datetime =
+                DateTime::Parse(referenceData(i, "DATE").accessString(), DateTime::Format::Standard);
+
+            const LLA lla = LLA(Angle::Degrees(latitude), Angle::Degrees(longitude), Length::Meters(altitude));
+
+            const Position position = {
+                lla.toCartesian(
+                    EarthGravitationalModel::WGS84.equatorialRadius_, EarthGravitationalModel::WGS84.flattening_
+                ),
+                Position::Unit::Meter,
+                Frame::ITRF()};
+
+            Instant instant = Instant::DateTime(datetime, Scale::UTC);
+
+            const Position sunPosition = sunFrameSPtr->getOriginIn(Frame::ITRF(), instant);
+
+            const Real density = nrlmsise.getDensityAt(position, instant, sunPosition);
+
+            const Real tolerance = 2.472e-14;
```

Comment:
I don't anticipate that the tolerance should change with different altitudes. To me it feels like a measure of how close our code is to their's. (let's find out! adding that other test now)
I don't have strong opinions for which style we use. To me using absolute values gives a good sense of how close you are to machine precision (1e-15), i.e. "exact" matching. But for things that aren't that close, I can definitely see a good case for percentages.

Reply:
So looks like you have a good point about other altitudes. For sweeping other altitudes, we need an artificially bigger tolerance to account for the lower altitudes. 

However, using a % tolerance doesn't work super well at higher altitudes (density 1e-14 and lower) because 1e-15 is machine precision, so we get a lot of variability in the trailing digits. So maybe it doesn't make sense to add it for the current tests.

But for the parameter sweeping test, I added something like this:
```
            // Check percent tolerance here for low altitudes
            const Real percentTolerance = 0.6;
            const Real percentError = 100.0*(std::abs(density - referenceDensity) / referenceDensity);

            // But still allow a pass if the absolute error is small (because we're close to machine precision)
            const Real absoluteTolerate = 1e-15;
            const Real absoluteError = std::abs(density - referenceDensity);

            EXPECT_TRUE( absoluteError < absoluteTolerate || percentError < percentTolerance ) << String::Format(
                "{} ≈ {} Δ {} [{}%] [T]", density.toString(), referenceDensity.toString(), absoluteError, percentError
            );
        }
```

What do you think?

Reply:
I think that would make sense yes 👍